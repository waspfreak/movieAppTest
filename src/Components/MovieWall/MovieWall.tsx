import React from 'react';
import { StyledMovieWall, StyledMovieWallContent, Style } from './style';

/**
 * MovieWall UI component for user interaction
 */
export interface MovieWallProps {
	header?: string;
	children: any;
}

export const MovieWall: React.FC<MovieWallProps> = ({
	header,
	children,
	...props
}) => {

	return (
		<Style {...props}>
			<h2 className='title'>{header}</h2>
			<StyledMovieWall>
				<StyledMovieWallContent>{children}</StyledMovieWallContent>
			</StyledMovieWall>
		</Style>
	);
};



export default MovieWall;
