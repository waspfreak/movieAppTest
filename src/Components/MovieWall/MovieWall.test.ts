import {snapshotTestFactory} from '../../utils/snapshotTestFactory';
//
import {MovieWall, MovieWallProps} from './MovieWall';
const props: MovieWallProps[] = [
	{
	header: "Title of your wall",
	children: "any component you need",
	},
];
snapshotTestFactory(MovieWall, props);
