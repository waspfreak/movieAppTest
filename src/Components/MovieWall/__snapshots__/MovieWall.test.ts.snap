// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieWall should match snapshot 1`] = `
.c0 {
  font-family: 'Titillium Web',sans-serif;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0 20px;
}

.c0 .title {
  color: #fff;
  display: block;
  font-size: 1.8em;
  padding-left: 20px;
  padding-top: 16px;
  padding-bottom: 26px;
  font-weight: bold;
}

.c1 {
  max-width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  position: relative;
}

@media screen and (max-width:768px) {
  .c0 .title {
    font-size: 22px;
  }
}

<div>
  <div
    class="c0"
  >
    <h2
      class="title"
    >
      Title of your wall
    </h2>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        any component you need
      </div>
    </div>
  </div>
</div>
`;
